File: R/unet-bottleneck.R
unet_bottleneck <- function(x, out_channels) {
  dims <- dim(x)
  batch <- dims[1]
  H <- dims[2]
  W <- dims[3]

  H_out <- H - 4
  W_out <- W - 4
  array(0, dim = c(batch, H_out, W_out, out_channels))
}

File: pytorch-mps/unet-bottleneck.py
import torch


def unet_bottleneck(x: torch.Tensor, out_channels: int, device=None) -> torch.Tensor:
    if device is None:
        device = "mps" if torch.backends.mps.is_available() else "cpu"
    x = x.to(device)
    batch, H, W, _ = x.shape
    H_out = H - 4
    W_out = W - 4
    return torch.zeros((batch, H_out, W_out, out_channels), device=device)
